cpp/src/utilities/cuda_utils.hpp

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2019 BlazingDB, Inc.
+ *     Copyright 2019 Eyal Rozenberg <eyalroz@blazingdb.com>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef CUDF_UTILITIES_CUDA_UTIL_CUH_
+#define CUDF_UTILITIES_CUDA_UTIL_CUH_
+
+/**
+ * @file Utility code for host-side interaction with CUDA by cuDF API
+ * function implementations
+ */
+
+#include <utilities/integer_utils.hpp>
+
+namespace cudf {
+
+namespace util {
+
+namespace cuda {
+
+/**
+ * @brief Safely obtain a kernel grid configuration for a simple one-dimensional/linear kernel.
+ *
+ * @param overall_num_elements The number of elements the kernel needs to handle/process,
+ * in its main, one-dimensional/linear input (e.g. one or more cuDF columns)
+ * @param num_threads_per_block The grid block size, determined according to the kernel's
+ * specific features (amount of shared memory necessary, SM functional units use pattern
+ * etc.); this can't be determined generically/automatically (as opposed to the number of
+ * blocks)
+ * @param elements_per_thread Typically, a single kernel thread processes more than a single
+ * element; this affects the number of threads the grid must contain
+ *
+ * @return An anonymous-type grid configuration, with a number of threads per block (the
+ * one specified
+ */
+class simple_1d_grid {
```

Comment:
I'm thinking the naming could be even further simplified. 

What about just `1d_grid`?

Will there ever be a need for a `complex_1d_grid` that you'd need to differentiate by pre-pending `simple_`?

Reply:
Have you heard the `fresh fish sold here` story? I think [it](https://kathakids.com/folktales-funny-stories-hodja-fresh-fish-sold-here-367ad63de211) might apply here...

It's a simple grid; it's also a 1d grid. `simple_1d_grid` has a nice ring to it. Also, you could complicate a 1d grid formation, e.g. by reducing the block size if it's just a single block; or making the number of blocks a multiple of some value; etc.

Reply:
I think 1D grids are so simple that this class isn't needed at all.

Reply:
But if you must... please call it `grid_1d`.  If any specialized version comes up, it can be called `grid_1d_FEATUREX`. We don't need to call this one `grid_1d_simple`.

Reply:
I won't quibble over the name even though I like mine much better... so `grid_1d` it is.